daffodil-lib/src/main/scala/org/apache/daffodil/util/Validator.scala

```diff
@@ -85,5 +82,22 @@ object Validator extends NoBindingFactoryAdapter {
     val documentSource = new StreamSource(document)
     validator.validate(documentSource)
   }
+
+  def initializeValidator(validator: javax.xml.validation.Validator, errHandler: ErrorHandler, resolver: DFDLCatalogResolver):
+    javax.xml.validation.Validator = {
+    validator.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true)
+    //
+    validator.setFeature("http://xml.org/sax/features/validation", true)
+
+    // If you enable the feature below, it seems to do no validation at all. Just passes.
+    //          validator.setFeature("http://apache.org/xml/features/validation/dynamic", true)
+
+    validator.setFeature("http://apache.org/xml/features/validation/schema", true)
+    validator.setFeature("http://apache.org/xml/features/validation/schema-full-checking", true)
+    validator.setErrorHandler(errHandler)
+    validator.setResourceResolver(resolver)
+    validator
```

Comment:
Is a validator a serializable object? Is it a stateful object? I am wondering if we can compute this once-only and save as a member of the SchemaSetRuntimeData, or if that isn't possible, compute it once-only at runtime and reuse it for multiple validations.

Reply:
There's a handful of bugs related to improving how we do full validation:

https://issues.apache.org/jira/browse/DAFFODIL-2386 - Full infoset validation should stream use SAX ValidationHandler
https://issues.apache.org/jira/browse/DAFFODIL-1749 - Store Xerces validator in DataProcessor so reloaded parsers can use full validation
https://issues.apache.org/jira/browse/DAFFODIL-1685 - Full validation should create and initialize the validator before parsing/unparsing begins

These issues can probably all be fixed as part of one change, but I'd reccomend it not being done as part of this. There's already alot going on in this PR.